index.js

```diff
@@ -454,5 +448,99 @@ module.exports = {
      */
     reset() {
         webpackConfig = new WebpackConfig(runtimeConfig);
-    }
+    },
+
+    /**
+     * Initialize the runtime environment.
+     *
+     * It can be used to directly manipulate the Encore API without
+     * executing the "./node_module/.bin/encore" utility.
+     *
+     * Encore.configureRuntimeEnvironment(
+     *     // Environment to use (dev, dev-server, production)
+     *     'dev-server',
+     *
+     *     // Same options you would use with the
+     *     // CLI utility with their name in
+     *     // camelCase.
+     *     {
+     *         https: true,
+     *         keepPublicPath: true
+     *     }
+     * )
+     *
+     * Be aware than using this method will also reset the current
+     * webpack configuration.
+     *
+     * @param {string} environment
+     * @param {object} options
+     * @returns {exports}
+     */
+    configureRuntimeEnvironment(environment, options = {}) {
+        runtimeConfig = parseRuntime(
+            Object.assign(
+                {},
+                require('yargs/yargs')([environment]).argv,
```

Comment:
I'm not sure I understand this. Is there a need to continue to read the command line options? The command line options could be *totally* different (since there is a different executable being run).

I see that parseRuntime() looks for `argv._[0]` for the environment/command. Is that what you're trying to supply here? If so, I think we could set that manually - e.g. `options._ = [environment]`. Or, we could re-work `parse-runtime.js` so that the `argv._[0]` is actually done in `encore.js` and `parse-runtime` looked like `function(command, options, cwd) {`.